variables.tf

```diff
@@ -1,135 +1,155 @@
   default     = false
 }
 
 variable "allow_merge_commit" {
   type        = bool
-  description = "Toggle to enable Merge Commits for the Repository"
+  description = "Toggle to enable Merge Commits for the Repository."
   default     = true
 }
 
 variable "allow_squash_merge" {
   type        = bool
-  description = "Toggle to enable Squash Merges for the Repository"
+  description = "Toggle to enable Squash Merges for the Repository."
   default     = true
 }
 
 variable "allow_rebase_merge" {
   type        = bool
-  description = "Toggle to enable Rebase Merges for the Repository"
+  description = "Toggle to enable Rebase Merges for the Repository."
   default     = true
 }
 
+variable "allow_auto_merge" {
+  type        = bool
+  description = "Toggle to enable auto-merging pull requests on the repository."
+  default     = false
+}
+
 variable "delete_branch_on_merge" {
-  description = "Toggle to automatically delete merged Branches for the Repository"
+  type        = bool
+  description = "Toggle to automatically delete merged Branches for the Repository."
   default     = false
 }
 
 variable "has_downloads" {
   type        = bool
-  description = "Toggle to enable (deprecated) GitHub Downloads for the Repository"
+  description = "Toggle to enable (deprecated) GitHub Downloads for the Repository."
   default     = false
 }
 
 variable "auto_init" {
   type        = bool
-  description = "Toggle to create an initial commit in the Repository"
+  description = "Toggle to create an initial commit in the Repository."
   default     = false
 }
 
 variable "gitignore_template" {
   type        = string
-  description = "Template to use for initial `.gitignore` file for the Repository"
-  default     = ""
+  description = "Template to use for initial `.gitignore` file for the Repository."
+  default     = null
 }
 
 variable "license_template" {
   type        = string
-  description = "Identifier to use for initial `LICENSE` file for the Repository"
-  default     = ""
+  description = "Identifier to use for initial `LICENSE` file for the Repository."
+  default     = null
 }
 
 variable "default_branch" {
   type        = string
-  description = "Name of the Default Branch of the Repository"
+  description = "Name of the Default Branch of the Repository."
   default     = "main"
 }
 
+variable "pages_branch" {
+  type        = string
+  description = "Name of the GitHub Pages Branch of the Repository."
+  default     = "gh-pages"
+}
+
 variable "archived" {
   type        = bool
-  description = "Toggle to archive the Repository (see notes in `README.md`)"
+  description = "Toggle to archive the Repository (see notes in `README.md`)."
   default     = false
 }
 
 variable "archive_on_destroy" {
   type        = bool
-  description = "Toggle to archive the Repository on destroy"
+  description = "Toggle to archive the Repository on destroy."
   default     = false
 }
 
 variable "pages" {
   type        = map(any)
```

Comment:
Is it possible to strongly type this? Would also require refactor of main.tf.
Example:
```
variable "pages" {
  type = object({
    branch = string
    path   = string
  })
  description = "Configuration block for GitHub Pages."
  default     = null
}
```